exercises/bob/instructions.md

```diff
@@ -0,0 +1,18 @@
+# Instructions
+
+Your task is to determine what Bob will reply to someone when they say something to him or ask him a question.
+
+Bob only ever answers one of five things:
+
+- **"Sure."**
+  This is his response if you ask him a question, such as "How are you?"
+- **"Whoa, chill out!"**
+  This is his answer if you YELL AT HIM (in all capitals).
+- **"Calm down, I know what I'm doing!"**
+  This is what he says if you yell a question at him.
+- **"Fine. Be that way!"**
+  This is how he responds if you address him without actually saying anything.
```

Comment:
As a student this was a frustrating part of trial-and-error. What is the difference between "anything else" and "without actually saying anything"? I had to look into the test cases to get an answer here, and that might not be obvious to new students.

Reply:
@vaeng Do you think we should say more about what "not saying anything" means here?

Reply:
Hm, rereading your comment, it sounds like probably we need to say more.

Without saying anything means that it’s silence.

In a novel, it would be a conversation like this.
“You did it, didn’t you?”
“…”
“Answer me, when I speak to you!”

In the examples it would be silence. In all the tests, it's represented by all spaces and tabs, I think.

Anything else would be anything that doesn't explicitly fall into the first four categories. So a normal sentence, or an emphatic sentence, or quoting some famous person or whatever.

Is this useful context?

Reply:
How about:
```suggestion
  This is how he responds if you respond with silence.
```

Reply:
Now I understand how it was meant, but back then I was really unhappy with the exercise. I have no real suggestion how to put the "white space check" into a non-programmer sentence though.  Elixir and C++ also check for carriage returns and new-lines, which might be confusing if you don't know about them.

It has been a while, but I remember the instructions to be confusing. The intended test-case seems to be "white space" and I think I tested cases like "...", "?" and other not-alphanumeric cases. Look at the [cpp](https://github.com/exercism/cpp/blob/eec57a6b68a39f91a7c2a6f9db55d8a973eb418e/exercises/practice/bob/bob_test.cpp#L86) version for example. That is just a smile and a question mark. That is - in my book - like not saying anything, but would count as a question.

Reply:
I don't love _respond with silence_ because it suggests that Bob said something first, which isn't necessarily the case. Maybe a kid at school walks up to him and grabs his shirt and pushes him against the wall and stares him down, without saying anything.

Let me think about this for a bit.